flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/split/NumericColumnSplitsGenerator.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.api.java.io.jdbc.split;
+
+import org.apache.flink.api.java.io.QueryParamInputSplit;
+
+/** 
+ * 
+ * This splits generator assumes that the query has a BETWEEN constraint on a numeric column.
+ * The generated input splits will be of size equal to the configured fetchSize (maximum),
+ * ranging from the min value up to the max 
+ * 
+ * */
+public class NumericColumnSplitsGenerator implements JDBCInputSplitsGenerator {
```

Comment:
I think ParameterValueProviders are very query specific. Not sure if it makes sense to provide default implementations. This class should be moved to the test scope, IMO.